pkg/providers/aws/provider_redis_test.go

```diff
@@ -44,13 +45,16 @@ type mockElasticacheClient struct {
 	wantErrList       bool
 	wantErrCreate     bool
 	wantErrDelete     bool
-	wantEmpty         bool
 	replicationGroups []*elasticache.ReplicationGroup
 	cacheSubnetGroup  []*elasticache.CacheSubnetGroup
 
 	// new approach for manually defined mocks
 	// to allow for simple overrides in test table declarations
-	modifyCacheSubnetGroupFn func(*elasticache.ModifyCacheSubnetGroupInput) (*elasticache.ModifyCacheSubnetGroupOutput, error)
+	modifyCacheSubnetGroupFn    func(*elasticache.ModifyCacheSubnetGroupInput) (*elasticache.ModifyCacheSubnetGroupOutput, error)
+	deleteCacheSubnetGroupFn    func(*elasticache.DeleteCacheSubnetGroupInput) (*elasticache.DeleteCacheSubnetGroupOutput, error)
+	describeCacheSubnetGroupsFn func(*elasticache.DescribeCacheSubnetGroupsInput) (*elasticache.DescribeCacheSubnetGroupsOutput, error)
+	describeCacheClustersFn     func(*elasticache.DescribeCacheClustersInput) (*elasticache.DescribeCacheClustersOutput, error)
+	describeReplicationGroupsFn func(*elasticache.DescribeReplicationGroupsInput) (*elasticache.DescribeReplicationGroupsOutput, error)
 }
 
 func buildMockElasticacheClient(modifyFn func(*mockElasticacheClient)) *mockElasticacheClient {
```

Comment:
I've noticed a lot of tests return `&elasticache.DescribeCacheSubnetGroupsOutput{}` when overriding `describeCacheSubnetGroupsFn`. That seems like a sane default function for a mock. So maybe we could do something like?

```suggestion
func buildMockElasticacheClient(modifyFn func(*mockElasticacheClient)) *mockElasticacheClient {
	mock := &mockElasticacheClient{
	        describeCacheSubnetGroupsFn: // function that returns &elasticache.DescribeCacheSubnetGroupsOutput{}, nil
	 }
	if modifyFn != nil {
		modifyFn(mock)
	}
	return mock
}
```